streaming/rdm_atomic.c

```diff
@@ -75,9 +75,23 @@ static const char* get_fi_op_name(enum fi_op op)
 	switch (op) {
 	case FI_MIN: return "min";
 	case FI_MAX: return "max";
+	case FI_SUM: return "sum";
+	case FI_PROD: return "prod";
+	case FI_LOR: return "lor";
+	case FI_LAND: return "land";
+	case FI_BOR: return "bor";
+	case FI_BAND: return "band";
+	case FI_LXOR: return "lxor";
+	case FI_BXOR: return "bxor";
 	case FI_ATOMIC_READ: return "read";
 	case FI_ATOMIC_WRITE: return "write";
 	case FI_CSWAP: return "cswap";
+	case FI_CSWAP_NE: return "cswap_ne";
+	case FI_CSWAP_LE: return "cswap_le";
+	case FI_CSWAP_LT: return "cswap_lt";
+	case FI_CSWAP_GE: return "cswap_ge";
+	case FI_CSWAP_GT: return "cswap_gt";
+	case FI_MSWAP: return "mswap";
```

Comment:
<code>fi_tostr</code> is available for both <code>FI_TYPE_ATOMIC_OP</code> and <code>FI_TYPE_ATOMIC_TYPE</code>. But when multiple <code>fi_tostr</code> calls are made in a single call (e.g. <code>snprintf</code> or <code>fprintf</code>), it doesn't return the correct string. @a-ilango and I found that, in libfabric/src/fi_tostr.c line 607, a static char * buf is returned which causes the issue. For example in the following case, printf outputs "op: FI_MIN, datatype: FI_MIN"
<code>
op_type = FI_MIN;
</code>
<code>
datatype = FI_FLOAT;
</code>
<code>
printf("op: %s, datatype: %s\n", fi_tostr(&op_type, FI_TYPE_ATOMIC_OP), fi_tostr(&datatype, FI_TYPE_ATOMIC_TYPE));
</code>

Is there any reason to make the return buf static which won't allow the above case? @shefty @jsquyres